### ISSUE TITLE
Investigate async workers

### ISSUE THREAD

**Issue by DeD1rk** [svthalia/concrexit#3458]
### What?
Async (ASGI) workers for unicorn could mabe help our server handle more load. But on the other hand, while switching to ASGI is probaly easy, really benefitting from it _might_ require many changes to our code (e.g. using async queries). But I'm not sure about that, and documentation is poor. It might just as well be the case that monkeypatching psycopg (e.g. with 'gevent') is all we need. We probably need to experiment a bit with that.

### Why?
I think our servers currently handle very little load because they are waiting on the database. That's a consequence of using sync (WSGI) workers. That shouldn't be a problem for sites that are less DB intensive, and e.g. use caching a lot, but for us it really seems like a problem.

### How?
Experiment, experiment, experiment.

**Comment by JobDoesburg** [comment]
I see a nice use case for Thadmin. With asgi, websockets can be implemented so we don't need to keep polling to see whether an order is paid, which will make Thadmin way more efficient

**Comment by DeD1rk** [comment]
I did a bit of investigation already, I tried running locally (with postgres db) with gevent/eventlet/uvicorn workers in gunicorn (and as far as I can tell with correct monkeypatching), but for some reason they all give reduced throughput compared to gunicorn's async worker. I also tried Daphne, which also performs poorly. 

I can't help the feeling that I'm doing something wrong because some people seem to have good experience with gevent, but there are no docs indicating what I could be doing wrong. 

It might be that we really don't have anything to gain unless we use async views.
